
diff --git a/install/PackageConfig.cmake.in b/install/PackageConfig.cmake.in
index a4e0c22..e26e080 100644
--- a/install/PackageConfig.cmake.in
+++ b/install/PackageConfig.cmake.in
@@ -30,14 +30,16 @@ if(NOT TARGET __zenohc_shared)
         set_property(TARGET __zenohc_shared PROPERTY IMPORTED_IMPLIB "@CMAKE_INSTALL_LIBDIR_EXPORTS@/@IMPLIB@")
     endif()
 endif()
-if(NOT TARGET __zenohc_static)
-    add_library(__zenohc_static STATIC IMPORTED GLOBAL)
-    add_library(zenohc::static ALIAS __zenohc_static)
-    target_link_libraries(__zenohc_static INTERFACE @NATIVE_STATIC_LIBS@)
-    set_target_properties(__zenohc_static PROPERTIES
-        IMPORTED_LOCATION "@CMAKE_INSTALL_LIBDIR_EXPORTS@/@STATICLIB@"
-        INTERFACE_INCLUDE_DIRECTORIES "@CMAKE_INSTALL_INCLUDEDIR_EXPORTS@"
-    )
+if(@ZENOHC_INSTALL_STATIC_LIBRARY@)
+    if(NOT TARGET __zenohc_static)
+        add_library(__zenohc_static STATIC IMPORTED GLOBAL)
+        add_library(zenohc::static ALIAS __zenohc_static)
+        target_link_libraries(__zenohc_static INTERFACE @NATIVE_STATIC_LIBS@)
+        set_target_properties(__zenohc_static PROPERTIES
+            IMPORTED_LOCATION "@CMAKE_INSTALL_LIBDIR_EXPORTS@/@STATICLIB@"
+            INTERFACE_INCLUDE_DIRECTORIES "@CMAKE_INSTALL_INCLUDEDIR_EXPORTS@"
+        )
+    endif()
 endif()

 if(NOT TARGET zenohc::lib)
